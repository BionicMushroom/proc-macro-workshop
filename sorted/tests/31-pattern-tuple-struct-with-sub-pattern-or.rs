use std::fmt::Display;

#[allow(dead_code)]
struct TestStruct(u32);

impl Display for TestStruct {
    #[allow(clippy::unnested_or_patterns, unused_parens)]
    #[sorted::check]
    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
        #[sorted]
        match self {
            TestStruct(0) => write!(f, "0"),
            TestStruct((5 | 6) | (1 | 2)) => write!(f, "1, 2, 5 or 6"),
            _ => write!(f, "other"),
        }
    }
}

fn main() {}
