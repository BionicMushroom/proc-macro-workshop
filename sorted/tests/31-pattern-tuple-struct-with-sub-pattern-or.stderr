error: `(1 | 2)` should sort before `(5 | 6)`
  --> tests/31-pattern-tuple-struct-with-sub-pattern-or.rs:13:34
   |
13 |             TestStruct((5 | 6) | (1 | 2)) => write!(f, "1, 2, 5 or 6"),
   |                                  ^^^^^^^
